src/content/actions/test/receive-profile.js

```diff
@@ -0,0 +1,118 @@
+import { assert } from 'chai';
+import sinon from 'sinon';
```

Comment:
yeah, I really like sinon, it makes it easy to test complex things. I agree we need to take care because it also makes it easy to test things we shouldn't test -- and maybe I did it here, I'll have a second look.

In the case of this PR, I want to test what happens with various HTTP results. There are several ways to do this:
* have a real server and have it return what we want
* use a fetch that uses node's http and https packages (for examle `isomorphic-fetch`), and mock them with `nock` (https://github.com/node-nock/nock)
* mock fetch with sinon (or jest's mock capability)

I decided to go with the latter because `sinon` is so much easier to work with than `nock` or an adhoc server (especially our server URL is hardcoded).